news/_posts/2020-06-18-about-bc-breaks.md

```diff
@@ -0,0 +1,400 @@
+
+Example of a soft BC break:
+```php
+public function transferPrices($product)
+{
+	return array(1 => '1', 2 => '2');
+}
+```
+becomes
+```php
+public function transferPrices($product)
+{
+	return array(1 => 1, 2 => 2);
+}
+```
+In this last snippet of code, the returning value is an array. Before the change, this is an array of strings while after the change it has become an array of integers. This small change might not throw an error in external code however it might affect their behavior and introduce bugs in a vicious manner.
+
+<hr>
+
+*Now let's have a look at different types of BC breaks*
+
+
+
+## Code BC breaks
+
+Code BC breaks are the most common and known BC breaks, and so far we have only been talking about them.
+
+The two examples used above, are code BC breaks.
+
+One thing worth mentioning is that a BC break code can happen in various (and subtle) ways. For example if `MyManager` extends a class `MyAbstractManager`, then a change in `MyAbstractManager` can create a BC break in `MyManager` by inheritance!
+
+Another subtle way of introducing a BC break is to update an existing PHP [interface](https://www.php.net/manual/en/language.oop5.interfaces.php). All classes implementing an interface must implement all required methods of the interface. If we add new methods into the interface, any class implementing the interface will immediately be considered faulty by PHP compiler, until the developer adds the new methods into it.
+
+### Deprecations
+
+There is a standard mechanism to handle code BC breaks: [deprecations](https://en.wikipedia.org/wiki/Deprecation). The idea of deprecation is to put inside the code a log message, that should be read by developers, to tell them what parts of the code will be removed soon, in the next major version. This way, developers can be informed early of the parts of the code they should not rely on in the future and adapt accordingly. This mechanism is [widely used in the software world](https://symfony.com/doc/current/contributing/code/bc.html) and has proven its effectiveness.
+
+So the full process goes like this:
+
+- developers decide to deprecate part of code A
+- developers put a deprecation message inside code A, warning users that "code A is deprecated and you should stop using it because it will soon be removed" however they keep it working for now
+- when next major version is released, the deprecated part of code A is gone
+
+You can see a very recent example in [Pull Request 19657](https://github.com/PrestaShop/PrestaShop/pull/19657/files#diff-f31f293b83e5d491f378ab486297a67bR196).
+
+### How PrestaShop maintainers handle this type of BC break
+
+We use the deprecations system to warn people as much as possible about incoming BC breaks.
```

Comment:
here it's ok to keep "we" because it's clear that "we" is the prestashop maintainers